### ISSUE TITLE
DeepExplainer instability on VGG16

### ISSUE THREAD

**Issue by slundberg** [shap/shap#225]
The code below shows a scenario where the SHAP values estimated by DeepExplainer don't sum up to the output of the model (which is an error). It seems to be specific to the mongoose image input, and I am opening this issue to track it until it is resolved.

```python
from keras.applications.vgg16 import VGG16
from keras.applications.vgg16 import preprocess_input, decode_predictions
import numpy as np
import shap
import keras.backend as K
import json

# load pre-trained model and choose two images to explain
model = VGG16(weights='imagenet', include_top=True)
X,y = shap.datasets.imagenet50()
to_explain = X[[39,41]]

# load the ImageNet class names
url = "https://s3.amazonaws.com/deep-learning-models/image-models/imagenet_class_index.json"
fname = shap.datasets.cache(url)
with open(fname) as f:
    class_names = json.load(f)

# explain how the input to the 7th layer of the model explains the top two classes
def map2layer(x, layer):
    feed_dict = dict(zip([model.layers[0].input], [preprocess_input(x.copy())]))
    return K.get_session().run(model.layers[layer].input, feed_dict)
mid_layer = 19
e = shap.DeepExplainer((model.layers[mid_layer].input, model.layers[-1].output), map2layer(preprocess_input(X[:10].copy()), mid_layer))
shap_values,indexes = e.shap_values(map2layer(to_explain[1:2], mid_layer), ranked_outputs=4)

model_out = model.predict(preprocess_input(to_explain[1:2].copy()))[0,indexes[0]]
shap_value_sums = np.array([shap_values[i].sum() for i in range(len(indexes[0]))])
assert np.abs(shap_value_sums + e.expected_value[indexes[0]] - model_out).sum() < 1e-2
```

**Comment by slundberg** [comment]
Looks like this is coming from numerical overflow in the softmax layer during attribution propagation. I'll push a fix once I get it implemented.

**Comment by slundberg** [comment]
Fixed now in 0.23.1
